### ISSUE TITLE
Allow MessagingUsers to be created from service accounts

### ISSUE THREAD

**Issue by rgodfrey** [EnMasseProject/enmasse#2054]
## why

Allow MessagingUsers to be created from service accounts.  This will allow applications running in Kubernetes to use their service account credentials to authenticate against the address space

## Proposal

Update the Keycloak plugin to recognize a new type of credential for a service account - from a SASL perspective it will need to be sent using PLAIN.  The plugin will find the user, see it is a service account based user, and then attempt to authenticate with Kubernetes using the service account name / password.

## Testing

Test that a messaging user can be created from a service account
Test that the credentials of the service account can be used to make a messaging connection
Test that using the credentials of a service account which is not a messaging user does not allow a messaging connection to be made
Test that using an incorrect password for the service account does not allow a successful connection

## Documentation

Update docs on creating messaging users

**Comment by grs** [comment]
Do service accounts have passwords? (I thought they did not and just had tokens generated).

**Comment by rgodfrey** [comment]
@grs sorry - was writing from the point of view of SASL where it is a password... from the point of view of the service account it is a token (although not an OAUTH one)

**Comment by grs** [comment]
makes sense (allowing password file to be specified in the connect.json supported by proton python, c++ and rhea would then be nice)

**Comment by rgodfrey** [comment]
@lulf it will need to be a new authentication type - we can't use the federated identity as (in general) the service accounts need not be being served by an oauth provider (I assume that is the case for standard kubernetes).  I'm thinking that the (Keycloak) user will be created with a new type of credential "serviceAccount" or whatever, and which (at most) will hold the details of the Kuberenetes server.  On getting a login attempt for a user with such a credential, Keycloak will simply send the "password" received as a token to the  Kubernetes server and see if it is authenticated (there's a question here as to whether it should check that the username matches the authenticated account that Kubernetes finds)

**Comment by rgodfrey** [comment]
@lulf that's the plan - yes
